build.gradle.kts

```diff
@@ -34,24 +31,42 @@ plugins {
 	id("org.ajoberstar.grgit")
 }
 
-operator fun String.invoke(): String {
-	return (rootProject.properties[this] as String?)!!
-}
+operator fun String.invoke(): String = rootProject.properties[this] as? String ?: error("Property $this not found")
 
 enum class ReleaseChannel(
 	val suffix: String? = null,
 	val releaseType: ReleaseType? = null,
-	val compress: Boolean,
+	val deflation: JarShrinkingType,
+	val classes: ClassShrinkingType,
+	val json: JsonShrinkingType,
 	) {
-	DEV_BUILD(suffix = "dev", compress = false),
-	PRE_RELEASE(suffix = "pre", compress = false),
-	RELEASE_CANDIDATE(suffix = "rc", compress = true),
-	RELEASE(releaseType = ReleaseType.STABLE, compress = true),
+	DEV_BUILD(
+		suffix = "dev",
+		deflation = JarShrinkingType.LIBDEFLATE,
```

Comment:
You can probably use max deflation for all builds unless it takes considerably longer or something.

Reply:
it's not necessarily longer in terms of how much time it takes, but in terms of scale it's about 90 times slower (from 0.005 to 0.43 seconds).

Reply:
0.43 seconds is negligible but up to you. I'm fine with max lossless compression for all builds.

Reply:
Alright, i'll switch it, but keep the option to switch back. It was 0.43 seconds on my computer, so on github servers it might be a little slower.